enhancements/sig-architecture/33-hub-templates-securing-sensitive-data/README.md

```diff
@@ -0,0 +1,268 @@
+implementation does make it so that the duplicate plaintext values that are
+encrypted in a policy for a particular managed cluster would appear the same
+since they would use the same IV and AES key. This added risk is minimal though
+since this can already be determined by looking at the original root policy and
+seeing the Hub policy templates referencing the same key (i.e. field name) of
+the data source (e.g. a Kubernetes secret).
+
+#### Key Generation and Storage
+
+The AES-256 key needs to be generated and stored both on the Hub cluster and the
+managed cluster. There would be a unique key per managed cluster. This would
+mean that if a single managed cluster is compromised, it does not mean that all
+encrypted values in the other managed clusters have also been compromised.
+
+The Policy Propagator should be responsible for generating the AES-256 key for
+each managed cluster. Each key would be stored in a secret called
+`policy-encryption-key` in each respective managed cluster namespace. The key
+would be stored on the secret’s `key` property. This secret should also have the
+`cluster.open-cluster-management.io/backup: 'policy'` label set so that the key
+is backed up during backup and restore situations. With this in place, the
+Policy Propagator is able to encrypt the data using the `protect` template
+function, however, the managed cluster also needs the key when it decrypts the
+data. The Policy Spec Sync controller should also sync the
+`policy-encryption-key` secret to the managed cluster in the managed cluster
+namespace by watching the managed cluster namespace for secrets on the Hub. It
+is also desired to allow customers to provide their own keys. This would be
+limited to AES-256 keys unless a customer requests otherwise.
+
+One thing to consider is if we want to add periodic automatic key rotations per
+managed cluster. This would add security, however, it would cause all policies
+to be reprocessed by the Policy Propagator. All policies that used the `protect`
+template function would need to be replicated to the managed clusters. This
+could cause a lot of churn in large scale scenarios. Additionally, there would
+be an odd period where the key is changed on the Hub and the managed cluster,
+but the recomputed policies with the new key would not be replicated yet, thus
+the Configuration Policy controller would fail to decrypt the values and mark
+all policies as non-compliant. There are some workarounds for this, but it is
+proposed to skip this for the upcoming release and provide documentation to
+manually rotate the key in the event a customer detects that it has been
+compromised. This will likely be made automatic in a future release.
+
+#### RBAC
+
+Since the encryption key is stored in a secret in the managed cluster namespace,
+it is important that non-privileged users do not have access to read this value.
+The default ACM roles of `open-cluster-management:view:<managed_cluster_name>`
```

Comment:
```suggestion
The default OCM roles of `open-cluster-management:view:<managed_cluster_name>`
```
Can you verify in OCM is the same set of roles?

Reply:
we do not have this settings in upstream, I think we should remove this part to avoid confusion